File: runtime/ruby/units/wasi/environ_sizes_get.rb
# requires: memory/iws
def wasi_environ_sizes_get(count_ptr, buf_size_ptr)
  @memory.iws(count_ptr, @env.size)
  @memory.iws(buf_size_ptr, @env.sum { |e| e.bytesize + 1 })
  ERRNO_SUCCESS
end

File: runtime/python/units/wasi/environ_sizes_get.py
# requires: memory/i32_store
def wasi_environ_sizes_get(self, count_ptr, buf_size_ptr):
    self.memory.i32_store(count_ptr, len(self.env))
    self.memory.i32_store(buf_size_ptr, sum(len(e) + 1 for e in self.env))
    return self.ERRNO_SUCCESS
